### DISCUSSION TITLE
Add attributes to specify Given, When, Then statements

### DISCUSSION THREAD

**Discussion by kuyer2022** [xunit/xunit#2664] [Ideas]
I think it would be quite helpful, if Given/When/Then statements could be provided in an attribute rather than specifying it as a method name. The method names are hard to read and we cannot put punctuation marks in it.

Also, the Test adapters should use those attributes to display those Given/When/Then statements in readable English.

For example, I wish my test method should be written as below, note the method name is a random string. The developer would not spend time generating a unique and appropriate method name.

```csharp
[Theory]
[Given("The apparel type is shirt"]
[When("Discount is  applied")]
[Then("It should be 20%")
[InlineData("shirt", 20)
public void YgyIY10BOB(string apparelType, int expectedResult)
{
    int discount = GetDiscount(apparelType);
    discount.should().Be(expectedResult);
}
```

**Comment by bradwilson** [comment]
Given/when/then isn't something that we explicitly (or implicitly) support or advocate.

There are two built-in ways to get a different display name for your test:

- Use [Theory(DisplayName="...")]
- Enable [methodDisplayOptions](https://xunit.net/docs/configuration-files#methodDisplayOptions) to transform "encoded" parts of the test method name into their appropriate symbols
